### PR TITLE
ARROW-3448: [Python] Add test for roundtrip of pandas DataFrame with column of lists of datetime objects

### PR DESCRIPTION
**Pull request by kszucs**
The issue has been already resolved, just updating the test examples.

### CODE REVIEW

File: python/pyarrow/tests/pandas_examples.py [hunk 1]
```diff
@@ -143,13 +143,27 @@ def dataframe_with_lists(include_index=False, parquet_compatible=False):
...
         (pa.date32(), date_data),
         (pa.date64(), date_data),
         (pa.time32('s'), time_data),
         (pa.time32('ms'), time_data),
-        (pa.time64('us'), time_data)
+        (pa.time64('us'), time_data),
+        (pa.timestamp('s'), datetime_data),
+        (pa.timestamp('ms'), datetime_data),
+        (pa.timestamp('us'), datetime_data),
+        (pa.timestamp('ns'), pandas_timestamp_data),
```
Comment (jorisvandenbossche) [inline python/pyarrow/tests/pandas_examples.py:166]: Do you get an error when you pass `datetime_data` here? (there are no nanoseconds here, so I would expect this to not matter)

**Review by jorisvandenbossche — COMMENTED** [review]
It's fine to add a test for this, but as I mentioned on the JIRA, this doesn't actually "fix" what's described in the title of the PR.

File: python/pyarrow/tests/pandas_examples.py (hunk 1 above)
Comment (kszucs) [reply python/pyarrow/tests/pandas_examples.py:166]: Yes:
```python
E   AssertionError: DataFrame.iloc[:, 12] (column name="timestamp[ns]_list") are different
E
E   DataFrame.iloc[:, 12] (column name="timestamp[ns]_list") values are different (80.0 %)
E   [index]: [0, 1, 2, 3, 4]
E   [left]:  [[2015-01-05T12:00:00.000000000, 2020-08-22T10:05:00.000000000], [2024-05-05T05:49:01.000000000, 2015-12-24T22:10:17.000000000], [1996-04-30T02:38:11.000000000], None, [1987-01-27T08:21:59.000000000, NaT]]
E   [right]: [[2015-01-05 12:00:00, 2020-08-22 10:05:00], [2024-05-05 05:49:01, 2015-12-24 22:10:17], [1996-04-30 02:38:11], None, [1987-01-27 08:21:59, None]]
```

**Comment by pitrou** [comment]
@jorisvandenbossche @kszucs What is the status of this PR? Should it be revived or closed?

**Comment by kszucs** [comment]
We can close this.

**Pull request closed without merging**